BankAccount.rb

```diff
@@ -0,0 +1,92 @@
+require 'csv'
+  # Create a module with the name Bank that will contain the class Account
+module Bank
+  # Create an Account class
+
+class Owner
+
+    attr_accessor :name
+
+    def initialize(name)
+    @name = name
+    #@balance = balance
+    puts "Hello, #{@name}."
+    end
+end
+
+class Account
+
+  attr_reader :account_id, :balance, :opendate
+  # set my deafualt balance to 100
+  def initialize(id, balance, opendate)
+      if balance < 0
+        raise ArgumentError, "Your balance is negative!"
+      end
+    @id = id
+    @balance = balance
+    @opendate = opendate
+  end
+
+  def display_account
+    # show balance of account at initial start
+    # show the account id
+    puts @balance
+  end
+
+  def withdraw(amount)
+    # accepts single parameters that represents the amount to be withdraw
+    # return updated balance
+    ## balance sets up the method that will subtract the amount from the balance
+    ## have the (-) subtracts from the set from below variable
+    if @balance - amount < 0
+     puts "You have a negative balance!"
+     puts "Please deposit money into your account."
+    else
+      @balance -= amount
+      puts @balance
+    end
+    return @balance
+  end
+
+
+  def deposit(amount)
+    # accept single parameter that represents the amount of money to be depostied
+    @balance += amount
+    # should return updated account balance
+    puts @balance
+  end
+
+  def self.all
+    # open a file and puts it in the variable back_csv
+    # read tells it to format the text.
+    # puts the whole thing in an array, puts the lines in little arrays.
+    bank_csv = CSV.read("./support/accounts.csv")
+    # displaying the content
+    print bank_csv
```

Comment:
I'd recommend removing the printing of the CSV because the method should return the contents of the data instead